### DISCUSSION TITLE
use debug and log option in production

### DISCUSSION THREAD

**Discussion by SabineMa** [ImageMagick/ImageMagick#3742] [Help]
Hi,

we are looking for a problem which occurs only on our production system. 
I am thinking about using the debug and log option of imagemagik in our production system for finding the reason.

I have the following questions:
1) Is it recommended NOT to do this or can I have at try with that? I am happy about your opinion.
2) where do I find the log file?
3) how can I put in the log file only that what I get from command line when I call the imagemagik command from command line? (I try with  and -debug ALL  and that is too much information :-) )

Sabine

**Comment by dlemstra** [comment]
Are you executing ImageMagick from the command line?

**Reply by SabineMa** [reply]
I call it from my application. I assume for IM it is like command line.

I just read about log.xml and am searching my hd for the files...

**Reply by SabineMa** [reply]
example (shortened): 

`"/usr/local/Cellar/imagemagick/7.0.10-35/bin/convert" -debug ALL -log "%u %m:%l %e"  -density 200   \( /Library/WebServer/Documents/reports//Buchungsliste.pdf -background white  -alpha remove -density 200 -strip -write /Library/WebServer/Documents/reports/Buchungsliste.png \)  null`

**Reply by SabineMa** [reply]
I think I found an easy way:

https://askubuntu.com/questions/420981/how-do-i-save-terminal-output-to-a-file

I want output only if an error occurs. So I can use 
`command 2>> output.txt`

`"/usr/local/Cellar/imagemagick/7.0.10-35/bin/convert" -debug ALL -log "%u %m:%l %e" -density 200 \( /Library/WebServer/Documents/reports//Buchungsliste.pdf -background white -alpha remove -density 200 -strip -write /Library/WebServer/Documents/reports/Buchungsliste.png \) null 2>> output.txt`

It creates a file with the error and attaches if there are several errors

then I don't need the log...
